web/src/utils/handleFileUpload.ts

```diff
@@ -0,0 +1,40 @@
+import { uploadFileToIPFS } from "./uploadFileToIPFS";
+import { uploadTransactionObject } from "./uploadTransactionObject";
+
+export const handleFileUpload = async (
+  escrowTitle: string,
+  deliverableText: string,
+  setIsFileUploading: (isFileUploading: boolean) => void,
+  setExtraDescriptionUri: (extraDescriptionUri: string) => void,
+  deliverableFile?: File
+) => {
+  try {
+    setIsFileUploading(true);
+    const transactionDetails = {
+      title: escrowTitle,
+      description: deliverableText,
+    };
+    setExtraDescriptionUri("");
+
+    if (deliverableFile) {
+      if (deliverableFile.type !== "application/pdf") {
+        alert("That type of file is not valid. Please upload a PDF file.");
+        setIsFileUploading(false);
+        return;
+      }
+
+      const fileResponse = await uploadFileToIPFS(deliverableFile);
+      const fileData = await fileResponse.json();
+      const fileHash = fileData.cids[0];
+      transactionDetails.extraDescriptionUri = fileHash;
```

Comment:
There was some discussion regarding using `ipfs://...` to conform with RFC3986, we will need to update `uploadFileToIpfs` to make it return the cids with that format.